### PR TITLE
Introduce `is distinct from` predicate operation

### PR DESCRIPTION
**Pull request by IzerOnadimQC**
Introduces an `is distinct from` predicate operation, which replaces the current `!=` operation. The current behaviour of `!=` is deprecated, and will be replaced in version 5.0 with an SQL-like behaviour, i.e. missing values will not be returned when using the predicate.

- [x] Changelog entry

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -2,11 +2,16 @@
...
 =========
 
-Plateau 4.2.3 (2023-11-??)
+Plateau 4.3.0 (unreleased)
 ==========================
 
+This minor release includes a slight API change to the behaviour of the ``!=`` predicate operation.
+
+* Add ``is distinct from`` predicate operation. This follows the behaviour of the SQL operation of the same name, i.e. it treats two null values as the same. Previously, this was also the behaviour of the ``!=`` operation.
+* The ``!=`` predicate operation now behaves the same as its SQL counterpart, i.e. it filters out null/missing values. The old behaviour of ``!=`` can now be acheived with ``is distinct from``.
```
Comment (IzerOnadimQC) [inline CHANGES.rst:11]: @jtilly @xhochy I had a couple of queries here.

The idea behind this change is to remove the inconsistency with SQL (which also matches the behaviour of our internal packages). However, this requires a breaking API change for no reason other than consistent naming. If you think that it's not worth changing the API, my other idea is to leave `!=` as is, and introduce another operation called `is not` which behaves the same as in SQL.

My other question is concerning versioning - based on what I know about semantic versioning, even a very small API change requires a change to the major version of the package - do you think I can leave the version as 4.3.0 or should I bump it to 5.0.0 (assuming we decide to go through with the API change)?

Also, as far as I can tell 4.2.2 was not actually released - perhaps I should merge that changelog entry with this one?

Comment (jtilly) [reply CHANGES.rst:11]: I think we should add the `is distinct from` operator in this PR and properly deprecate the `!=` behavior (which will then eventually be part of a 5.0 release).

Comment (IzerOnadimQC) [reply CHANGES.rst:11]: Ok sounds good - in that case, the `is distinct from` and `!=` operations will have exactly the same behaviour until 5.0, I suppose?

Comment (jtilly) [reply CHANGES.rst:11]: Yes, exactly.

**Review by jtilly — APPROVED** [review]

File: CHANGES.rst [hunk 2]
```diff
@@ -2,11 +2,14 @@
 Changelog
 =========
 
-Plateau 4.2.3 (2023-11-??)
+Plateau 4.3.0 (unreleased)
```
Comment (xhochy) [inline CHANGES.rst:5]: Do you want to release today or next year?

**Review by xhochy — APPROVED** [review]

File: CHANGES.rst (hunk 2 above)
Comment (IzerOnadimQC) [reply CHANGES.rst:5]: I'm happy to do a release today if you're OK with that.

**Comment by xhochy** [comment]
@IzerOnadimQC Can you fix the CI error (data missing for 14.0.2)? Then we can merge&release.

**Pull request merged**